ruby/lib/clone_handler.rb

```diff
@@ -0,0 +1,122 @@
+require 'json'
+require 'net/http'
+
+class CloneHandler
+  PER_PAGE = 30.freeze
+  BASE_URL = 'https://api.github.com'
+
+  def self.get_username
+    Kernel.puts 'Please enter your Github username'
+    STDOUT.flush
+    STDIN.gets.chomp
+  end
+
+  def initialize(username)
+    @username = username
+    @errors = []
+  end
+
+  def clone
+    validate_clonable
+    if errors?
+      print_errors
+      return false
+    end
+
+    Kernel.system(final_command)
+  end
+
+  private
+
+  attr_reader :username, :errors
+
+  def final_command
+    @final_command ||= clone_commands.flatten.join(' & ')
+  end
```

Comment:
I still do not like this "final_command" name
`string_to_clone_all_repos_as_bash_jobs` 🤣

Reply:
`clone_all_command`? Open to easy to read ideas.